python/dgl/dataloading/labor_sampler.py

```diff
@@ -0,0 +1,176 @@
+#
+#   Copyright (c) 2022 by Contributors
+#
+#   Licensed under the Apache License, Version 2.0 (the "License");
+#   you may not use this file except in compliance with the License.
+#   You may obtain a copy of the License at
+#
+#       http://www.apache.org/licenses/LICENSE-2.0
+#
+#   Unless required by applicable law or agreed to in writing, software
+#   distributed under the License is distributed on an "AS IS" BASIS,
+#   WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+#   See the License for the specific language governing permissions and
+#   limitations under the License.
+#
+#   Based off of neighbor_sampler.py
+#
+
+"""Data loading components for labor sampling"""
+from ..base import NID, EID
+from ..transforms import to_block
+from .base import BlockSampler
+from ..random import choice, seed
+from .. import backend as F
+from collections import defaultdict
```

Comment:
For code style conformation please refer to https://docs.google.com/document/d/1iHyj7zlmygKSk5gBPsqIqL5ASPzJSPREaNT_QdsiYA4/edit